packages/domain/lib/use_case/palindrome_use_case.dart

```diff
@@ -1,19 +1,19 @@
 import 'package:data/repository/data_repository.dart';
-import 'package:domain/use_case/use_case.dart';
 
+part 'use_case.dart';
 
-class PalindromeUseCaseImpl implements UseCase {
-  PalindromeUseCaseImpl(this._palindromeRepositoryImpl);
+class PalindromeUseCaseImpl implements UseCase<String, Future<bool>> {
+  PalindromeUseCaseImpl(this._repository);
 
-  final PalindromeRepositoryImpl _palindromeRepositoryImpl;
+  final PalindromeRepositoryImpl _repository;
```

Comment:
u should work with PalindromeRepository (abstraction) here, but not implementation

Reply:
fixed